src/wp-includes/blocks.php

```diff
@@ -1358,3 +1358,153 @@ function _wp_multiple_block_styles( $metadata ) {
+		'update_comment_meta_cache' => false, // We lazy-load comment meta for performance.
+	);
+
+	if ( ! empty( $block->context['postId'] ) ) {
+		$comment_args['post_id'] = (int) $block->context['postId'];
+	}
+
+	if ( get_option( 'thread_comments' ) ) {
+		$comment_args['hierarchical'] = 'threaded';
+	} else {
+		$comment_args['hierarchical'] = false;
+	}
+
+	$per_page     = get_option( 'comments_per_page' );
+	$default_page = get_option( 'default_comments_page' );
+
+	if ( $per_page > 0 ) {
+		$comment_args['number'] = $per_page;
+
+		$page = (int) get_query_var( 'cpage' );
+		if ( $page ) {
+			$comment_args['paged'] = $page;
+		} elseif ( 'oldest' === $default_page ) {
+			$comment_args['paged'] = 1;
+		} elseif ( 'newest' === $default_page ) {
+			$comment_args['paged'] = (int) ( new WP_Comment_Query( $comment_args ) )->max_num_pages;
+		}
+		// Set the `cpage` query var to ensure the previous and next pagination links are correct
+		// when inheriting the Discussion Settings.
+		if ( 0 === $page && isset( $comment_args['paged'] ) && $comment_args['paged'] > 0 ) {
+			set_query_var( 'cpage', $comment_args['paged'] );
+		}
+	}
+
+	return $comment_args;
+}
+
+/**
+ * Helper function that returns the proper pagination arrow html for
+ * `CommentsPaginationNext` and `CommentsPaginationPrevious` blocks based on the
+ * provided `paginationArrow` from `CommentsPagination` context.
+ *
+ * It's used in CommentsPaginationNext and CommentsPaginationPrevious blocks.
+ *
+ * @since 6.0.0
+ *
+ * @param WP_Block $block           Block instance.
+ * @param string   $pagination_type Type of the arrow we will be rendering.
+ *                                  Default 'next'. Accepts 'next' or 'previous'.
+ *
+ * @return string|null Returns the constructed WP_Query arguments.
+ */
+function get_comments_pagination_arrow( $block, $pagination_type = 'next' ) {
+	$arrow_map = array(
+		'none'    => '',
+		'arrow'   => array(
+			'next'     => '→',
+			'previous' => '←',
+		),
+		'chevron' => array(
+			'next'     => '»',
+			'previous' => '«',
+		),
+	);
+	if ( ! empty( $block->context['comments/paginationArrow'] ) && ! empty( $arrow_map[ $block->context['comments/paginationArrow'] ][ $pagination_type ] ) ) {
+		$arrow_attribute = $block->context['comments/paginationArrow'];
+		$arrow           = $arrow_map[ $block->context['comments/paginationArrow'] ][ $pagination_type ];
+		$arrow_classes   = "wp-block-comments-pagination-$pagination_type-arrow is-arrow-$arrow_attribute";
+		return "<span class='$arrow_classes'>$arrow</span>";
+	}
+	return null;
+}
+
+/**
+ * Workaround for getting discussion settings as block editor settings
+ * so any user can access to them without needing to be an admin.
+ *
+ * @since 6.0.0
+ *
+ * @param array $settings Default editor settings.
+ *
+ * @return array Filtered editor settings.
+ */
+function extend_block_editor_settings_with_discussion_settings( $settings ) {
+
+	$settings['__experimentalDiscussionSettings'] = array(
```

Comment:
This needs to go somewhere before this line:

https://github.com/WordPress/wordpress-develop/blob/b6394c2e9fcdea317ff350d59e985a40e02edd41/src/wp-includes/block-editor.php#L407